code/game/objects/items/devices/drop_targeter/_droptargeter.dm

```diff
@@ -0,0 +1,157 @@
+	var/turf/drop_turf // If you want to turn the area the drop drops in, to a certain turf. I recommend using airless tiles, otherwise items will be strewn about
+
+/obj/item/device/orbital_dropper/attack_self(mob/user)
+	zoom(user, tileoffset, viewsize)
+
+/obj/item/device/orbital_dropper/attack(mob/living/M, mob/user)
+	laser_act(M, user)
+
+/obj/item/device/orbital_dropper/afterattack(var/atom/target, var/mob/living/user, flag, params)
+	if(flag)	//we're placing the targetter on a table or in backpack
+		return
+	laser_act(target, user)
+
+/obj/item/device/orbital_dropper/proc/laser_act(var/atom/target, var/mob/living/user)
+	if(has_dropped >= drop_amount)
+		to_chat(user, span("warning", "You can't use this device again!"))
+		return
+
+	var/turf/targloc = get_turf(target)
+	if(!emagged)
+		for(var/turf/t in block(locate(targloc.x+2,targloc.y+2,targloc.z), locate(targloc.x-2,targloc.y-2,targloc.z)))
+			if (!istype(t.loc, /area/mine))
+				to_chat(user, span("warning", "You can't do this so close to the station, point the laser further into the mine!"))
+				return
+			if (!isfloor(targloc))
+				to_chat(user, span("warning", "You cannot request a drill on unstable flooring!"))
+				return
+	if(!(user in (viewers(paint_distance, target))) )
+		to_chat(user, span("warning", "You can't paint the target that far away!"))
+		return
+	if (!user.IsAdvancedToolUser())
+		to_chat(user, span("warning", "You don't have the dexterity to do this!"))
+		return
+	add_fingerprint(user)
+
+	//laser pointer image
+	icon_state = "drillpointer_on"
+	var/list/showto = list()
+	for(var/mob/M in viewers(targloc))
+		if(M.client)
+			showto += M.client
+	var/image/I = image('icons/obj/projectiles.dmi', targloc, "red_laser", 10)
+	I.pixel_x = target.pixel_x + rand(-5,5)
+	I.pixel_y = target.pixel_y + rand(-5,5)
+
+	to_chat(user, span("notice", "You paint the target at [target]."))
+
+	var/obj/item/device/radio/intercom/announcer = new /obj/item/device/radio/intercom(null)
+	announcer.config(list("Common" = FALSE, "Entertainment" = FALSE, "Response Team" = FALSE, "Science" = FALSE, "Command" = FALSE, "Medical" = FALSE, "Engineering" = FALSE, "Security" = FALSE, "Supply" = FALSE, "Service" = FALSE, "Mercenary" = FALSE, "Raider" = FALSE, "Ninja" = FALSE, "AI Private" = FALSE))
+	if(announcer)
+		if(!emagged)
+			announcer.autosay(drop_message, announcer_name, announcer_channel)
+		else
+			announcer.autosay(drop_message_emagged, announcer_name, "Common")
+
+	has_dropped++
+	addtimer(CALLBACK(GLOBAL_PROC, /proc/explosion, targloc, 1, 2, 4, 6), 100) //YEEHAW
+	addtimer(CALLBACK(src, .proc/drill_drop, targloc), 105)
+
+	flick_overlay(I, showto, 20) //2 seconds of the red dot appearing
+	icon_state = "drillpointer"
+
+/obj/item/device/orbital_dropper/proc/drill_drop(var/turf/target)
+	var/turf/east_turf
+	var/turf/south_east_turf
+	var/turf/south_turf
+	var/turf/south_west_turf
+	var/turf/west_turf
+	var/turf/north_west_turf
+	var/turf/north_turf
+	var/turf/north_east_turf
```

Comment:
It'd be far cleaner and better to assign all of these in the line where they're actually used.

Reply:
Also these should probably be a single assoc list of lists that gets iterated over with a loop.